kernel/arch/riscv64/Hat.cpp

```diff
@@ -0,0 +1,250 @@
+#include <arch/Hat.h>
+#include <memory/Pmm.h>
+#include <debug/Log.h>
+#include <Memory.h>
+#include <Maths.h>
+
+namespace Npk
+{
+    enum PageSizes
+    {
+        _4K = 1,
+        _2M = 2,
+        _1G = 3,
+        _512G = 4,
+    };
+
+    struct PageTable
+    {
+        uint64_t entries[512];
```

Comment:
why not use a define for the number of entries? and reuse it?

Reply:
I will add that